### DISCUSSION TITLE
Shiny can not show all region options

### DISCUSSION THREAD

**Discussion by eeinnew** [teamalgoritma/community#92] [Q&A] [answered]
Hello, I have some error when I am about to plot in shiny
<img width="912" alt="image" src="https://user-images.githubusercontent.com/81936292/167740804-b0d6c389-bcce-4c55-b14a-c55ea5448012.png">
<img width="960" alt="image" src="https://user-images.githubusercontent.com/81936292/167740835-b85a8584-5bfe-40b5-ab87-6bcb1910f689.png">
I have changed the region filter as what the filter input, but in the input we can only choose Indonesia... while it suppose to have a list of country around the world

**Comment by inytss** [comment, marked as answer]
Dear @eeinnew , 
Can you please give me the code on your UI related to input$region? And please attach too the glimpse of data you used.
Thank you

**Reply by eeinnew** [reply]
``` 
      tabItem(tabName = "business_data",
              box(
                title = "Business data",
                width = 12,
                selectInput(inputId = "b_data",
                            label = "Choose the country",
                            choices = unique(Business_clean$Country_Name),
                            selected = "negara")
              ),
              fluidRow(
                box(width = 4,
                    plotlyOutput(outputId = "bus_dt1")),
                box(width = 4,
                    plotlyOutput(outputId = "bus_dt2")),
                box(width = 4,
                    plotlyOutput(outputId = "bus_dt3")),
              ),
              fluidRow(
                box(width = 6,
                    plotlyOutput(outputId = "bus_dt4")),
                box(width = 6,
                    plotlyOutput(outputId = "bus_dt5"))
              )
      )

**Reply by eeinnew** [reply]
this is my plot bus_dt1 in server
```
 output$bus_dt1 <- renderPlotly({
    Business_clean <- Business %>% 
      filter(Country_Name == input$b_data , 2005 < Time, Time < 2021 ) %>% 
      select(-Country_Code)
    BC_exp <- Business_clean %>% 
      mutate(label = glue("Country : {Country_Name}
                      Manufacture Exported : {Manufactures_exports_...}%"))
    BP_exp <- ggplot(BC_exp, aes(x = Time , y= Manufactures_exports_... , text = label)) +
      geom_col(aes(fill = Manufactures_exports_...)) +
      scale_fill_gradient(low = "lightpink1", high ="slateblue4") +
      geom_line(aes(group=1), colour="slateblue1")+
      geom_point(size=1.5, colour="slateblue2") +
      labs(title = "Manufactures exports (% of merchandise exports (2006-2020))",
           x = "Year",
           y = "Percentage") +
      theme_minimal() +
      theme(legend.position = "none", 
            plot.title = element_text(hjust = 0.5))
    ggplotly(BP_exp, tooltip = "text")
    
  })

**Comment by eeinnew** [comment]
![image](https://user-images.githubusercontent.com/81936292/167986188-33da429e-a8a1-494f-b6ef-9861a4449064.png)
and my global looks like this.
below I attach my business data
[Business.csv](https://github.com/teamalgoritma/community/files/8674795/Business.csv)

**Reply by eeinnew** [reply]
actually, yesterday I asked about maps plotting, but the next day I try it, suddenly the options of the region appears, but I do not know why, so I plot the second data but the same problem occurs

**Reply by inytss** [reply]
Dear @eeinnew , 
I'm sorry I don't get the point of your questions. To make it clear, I want to ask some questions related to your problem. Is your problem in plotting the map data with the object in server is mapworld? Or in plot bus_dt1? Because if I see your code in mapworld object, you use the mapdata1 dataframe object, but in plot bus_dt1 you use the Business dataframe object.

**Reply by eeinnew** [reply]
sorry for confusing, I did have problem with the map yesterday, but it has been settled.
However, the same problem occur in my other plot (currently, is business data) so the code I sent above is bus_dt, in which the country in the select only Indonesia

**Reply by inytss** [reply]
Oh I see. If I see in your read data code, you use `stringAsFactors = T` parameter, so it means that every columns with string format will save into factor type. If you don't mind, you can try in your console to run this code.
```
unique(Business_clean$Country_Name)

levels(Business_clean$Country_Name)
```

If the output of `levels(Business_clean$Country_Name)` appearing more than one country as it should be, you better use those code to fill in `choices` parameter in `selectInput()`. Since the columns stored as factor type it better `levels()` than `unique()` to point out the categories of the data. `unique()` better used in character type data.

**Comment by eeinnew** [comment]
okay, it works, thankyou
